core/src/main/scala/org/apache/spark/util/SignalUtils.scala

```diff
@@ -31,14 +31,51 @@ import org.apache.spark.internal.Logging
 /**
  * Contains utilities for working with posix signals.
  */
-private[spark] object Signaling extends Logging {
+private[spark] object SignalUtils extends Logging {
+
+  private var registered = false
+
+  /** Register a signal handler to log signals on UNIX-like systems. */
+  def registerLogger(log: Logger): Unit = synchronized {
```

Comment:
think about it for a second: with your definition we should probably have ~ 1 million files in spark, since every function is doing something slightly different.